src-frontend/app/controllers/procedure.js

```diff
@@ -1,28 +1,48 @@
 import Ember from 'ember';
 
-export default Ember.Controller.extend({
+export default Ember.ObjectController.extend({
   pages: function() {
-    return this.get('model.pages').sortBy('index');
+    return this.get('model.pages').sortBy('displayIndex');
   }.property('model.pages'),
 
   actions: {
+    addElement: function(page, selectedIndex) {
+      var targetIndex = selectedIndex + 1;
+      var elements = page.get('elements');
+      var elementsToBePushed = elements.filter(function(element) {
+        return element.get('displayIndex') >= targetIndex;
+      });
+
+      elementsToBePushed.forEach(function(element) {
+        element.incrementProperty('displayIndex');
+        element.save();
```

Comment:
I was told that Ember batches them up in the current run loop. But I'm still getting weird behaviour if I cascade a bunch of elements. I'll add a TODO